app/components/Connect4/lib/matches/diagonal.js

```diff
@@ -38,36 +38,29 @@ function isTopRight(grid) {
 
     found = 0;
     foundPiece = 0;
-    col = baseCol;
+    col = baseCol - 1; // Subtracting 1 to compensate for incrementing col at
+                       // the beginning of the loop
 
-    // Here we work our way up the current diagonal
+    // Here we work our way *UP* the current diagonal
     for (let row = 0; row < numRows; row++) {
+      col++;
 
       // Ensure that the given column and row are on the board
       if (col >= 0 && col < numCols && row < numRows) {
 
         let piece = grid[col][row];
 
-        // Does nothing if piece is zero
-        if (!piece) {
```

Comment:
You can't do nothing here. `found_piece` needs to be set to the current piece or counting will be off. Before this PR, non-contiguous lines were triggering a win. In other words, when you have on a diagonal: red, red, empty, red, red... the count would not be reset on the empty cell, and thus the count would reach 4.